pkg/reconcilermanager/controllers/rootsync_controller_test.go

```diff
@@ -70,6 +72,15 @@ const (
 
 var rootReconcilerName = core.RootReconcilerName(rootsyncName)
 
+func role(t *testing.T, name string, opts ...core.MetaMutator) *rbacv1.Role {
+	t.Helper()
+
+	result := fake.RoleObject(opts...)
+	result.Name = name
```

Comment:
nit: i think there's a name mutator. that would allow this to be inlined.

Reply:
There is, but I'm not sure of a cleaner way to combine that into a single variadic argument

Reply:
`append(Name(name), opts...)...`

Reply:
I feel that's less readable than the alternative here, but I don't feel strongly

Reply:
i don't disagree. just a cute way to make it short. no biggie.